x/core/keeper/gas_meter.go

```diff
@@ -63,9 +62,82 @@ func (k Keeper) DistributionsFromGasMeter(ctx sdk.Context, reqID string, reqHeig
 	return dists
 }
 
+func (k Keeper) ProcessDistributions(ctx sdk.Context, dists []types.Distribution, dr *types.DataRequest, minimumStake math.Int) error {
+	// Distribute in order.
+	denom, err := k.stakingKeeper.BondDenom(ctx)
+	if err != nil {
+		return err
+	}
+
+	remainingEscrow := dr.Escrow
+
+	// TODO Events
+	var amount math.Int
+	for _, dist := range dists {
+		switch {
+		case dist.Burn != nil:
+			amount = math.MinInt(dist.Burn.Amount, remainingEscrow)
+			err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(sdk.NewCoin(denom, amount)))
+			if err != nil {
+				return err
+			}
+
+		case dist.DataProxyReward != nil:
+			amount = math.MinInt(dist.DataProxyReward.Amount, remainingEscrow)
+			payoutAddr, err := sdk.AccAddressFromBech32(dist.DataProxyReward.PayoutAddress)
+			if err != nil {
+				// Should not be reachable because the address has been validated.
+				return err
+			}
+			err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, payoutAddr, sdk.NewCoins(sdk.NewCoin(denom, amount)))
+			if err != nil {
+				return err
+			}
+
+		case dist.ExecutorReward != nil:
+			amount = math.MinInt(dist.ExecutorReward.Amount, remainingEscrow)
+			staker, err := k.GetStaker(ctx, dist.ExecutorReward.Identity)
+			if err != nil {
+				return err
+			}
+
+			// Top up staked amount to minimum stake.
+			topup := math.ZeroInt()
+			if staker.Staked.LT(minimumStake) {
+				topup = math.MinInt(minimumStake.Sub(staker.Staked), amount)
+				staker.Staked = staker.Staked.Add(topup)
+				remainingEscrow = remainingEscrow.Sub(topup)
+			}
+			staker.PendingWithdrawal = staker.PendingWithdrawal.Add(amount.Sub(topup))
+
+			err = k.SetStaker(ctx, staker)
+			if err != nil {
+				return err
+			}
+		}
+
+		remainingEscrow = remainingEscrow.Sub(amount)
+	}
```

Comment:
We should be cautious with the errors in this loop. The contract would never error in this loop and instead would skip over the distrubtion and move on to the next one. The returns here, if I understand correctly, would cause this not to happen.